auth/default_authorizer.cc

```diff
@@ -181,23 +182,24 @@ default_authorizer::modify(
         permission_set set,
         const resource& resource,
         std::string_view op) const {
-    return do_with(
-            format("UPDATE {}.{} SET {} = {} {} ? WHERE {} = ? AND {} = ?",
-                    _auth_ks_name,
-                    PERMISSIONS_CF,
-                    PERMISSIONS_NAME,
-                    PERMISSIONS_NAME,
-                    op,
-                    ROLE_NAME,
-                    RESOURCE_NAME),
-            [this, &role_name, set, &resource](const auto& query) {
-        return _qp.execute_internal(
+    const sstring query = format("UPDATE {}.{} SET {} = {} {} ? WHERE {} = ? AND {} = ?",
+            _auth_ks_name,
+            PERMISSIONS_CF,
+            PERMISSIONS_NAME,
+            PERMISSIONS_NAME,
+            op,
+            ROLE_NAME,
+            RESOURCE_NAME);
+    if (legacy_mode(_qp)) {
+        co_return co_await _qp.execute_internal(
                 query,
                 db::consistency_level::ONE,
                 internal_distributed_query_state(),
                 {permissions::to_strings(set), sstring(role_name), resource.name()},
                 cql3::query_processor::cache_internal::no).discard_result();
-    });
+    }
+    co_return co_await announce_mutations(_qp, _group0_client, query,
+        {permissions::to_strings(set), sstring(role_name), resource.name()}, &_as);
```

Comment:
`query_processor` already knows how to execute some statements with group0 guard (see `execute_maybe_with_guard`) and even retry the command. When I added the code I wanted to make it generic so that it will work not only with ddl, but,  potentially, with auth as well, but was told that it is too early. So may be now is the time? What should be done to achieve that is to make `modification_statement` /`select_statement` to, somehow, recognize that it is an auth statement and set `needs_guard` flag and also to change execute path to use raft instead of the `storage_proxy` for those kind of statements.

Reply:
@avikivity @kostja you were those that wanted to make `query_processor` code schema statements specific.

Reply:
ok, that's a good idea I didn't think of before

Reply:
bouncing to shards 0 added, I think needs_guard = true additions are enough because I no longer allow bare inserts to system_auth_v2 tables, all need to be behind role & family statements. And reads are eventually consistent (as per auth-v2 design doc)

Reply:
But why can't ROLE statements be handled on the query processor level? Read statements may skip setting `needs_guard` flag.

Reply:
If I understand you question correctly, I wrote more about it in https://github.com/scylladb/scylladb/pull/16578#discussion_r1483086524

Reply:
I agree with handling in the query processor but I am happy with it happening in a follow up as well.

I just noticed that view update backlog is not moved to raft :(, so there is still work to do :(